packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -176,6 +181,11 @@ export default compose( [
 		const parentId = getBlockRootClientId( clientId );
 		const parentBlock = getBlock( parentId );
 
+		// Fix for android floating toolbar - show toolbar in parent for the first of inner block if is selected
+		const blockClientIds = getBlockOrder( clientId );
+		const selectedID = getSelectedBlockClientId();
+		const showFloatingToolbarForChild = blockClientIds && blockClientIds.length && selectedID === blockClientIds[ 0 ];
```

Comment:
If I understand correctly, this is assuming that the first block on a block list is always at the top, which might work for the current cases, but it will probably break in the future. Not a blocker for the fix (unless we can find an example of something that breaks), but we should start thinking about better ways to detect which blocks need to relocate their toolbar

Reply:
Yeah, we did this assumption as it works in all cases in the current state. I left a comment in the code why we do this, so it should be clear for the future